### PR TITLE
Modularization and more organized project structure

### PR DESCRIPTION
**Pull request by chessl**
First PR of a series of changes proposed in #15, sorry for such delay. Modularization for portability and accessibility. Reorganizes project structure, and switches to `rollup` as the builder to provide both `cjs` and `umd` outputs and better workflow for future development.

### CODE REVIEW

File: src/index.js [hunk 1]
```diff
@@ -0,0 +1,517 @@
...
+      }
+    }
+
+    if (scopedCSSOptions['applyToClass']) {
+      applyMode = applyToClass;
+    }
+  }
+
+  // nb. watch for attributeFilter: ['scoped'] to detect a CSS rule changing at runtime
+  const options = {childList: true, subtree: true, attributes: true, attributeFilter: ['scoped']};
```
Comment (samthor) [inline src/index.js:511]: not important for now, but I wonder if this actually detects content changes in style tags, or just the `scoped` attribute toggling...

**Review by samthor — COMMENTED** [review]
In general this is fine.

But I wonder if we can remove the dist code from the repo and just publish it to NPM?

File: rollup.umd.config.js [hunk 2]
```diff
@@ -0,0 +1,23 @@
+import { uglify } from 'rollup-plugin-uglify';
+
+const umdConfig = {
```
Comment (kuitos) [inline rollup.umd.config.js:3]: Maybe not just umd output? ESM and cjs are all necessary imo.

File: test.js [hunk 3]
```diff
@@ -14,7 +14,7 @@
  * the License.
  */
 
-import './scoped.js';
+import './dist\/scoped.js';
```
[the remark this answers is not in the document]
Comment (chessl) [reply test.js:17]: The testing framework used to give me a weird error until this forward-slash was escaped, but at the moment it seems like the error cannot be reproduced anymore. It's removed now.

File: src/index.js (hunk 1 above)
Comment (chessl) [reply src/index.js:511]: I've played it around by randomly changing a style tag's body and it works for me, will add a few tests to further confirm it.

**Comment by chessl** [comment]
You are absolutely right, they are totally unnecessary, I've just removed them.

File: rollup.umd.config.js (hunk 2 above)
Comment (chessl) [reply rollup.umd.config.js:3]:
```
"build:cjs": "rollup -i src/index.js -o lib/index.js -f cjs",
"build:cjs-polyfill": "rollup -i src/polyfill.js -o lib/polyfill.js -f cjs",
```

cjs output is already supported, will probably add esm output support in a separate pr.

File: package.json [hunk 4]
```diff
@@ -2,13 +2,32 @@
...
+    "dist"
+  ],
   "author": "Sam Thorogood <sam.thorogood@gmail.com>",
   "license": "Apache-2",
   "scripts": {
-    "test": "headless-test --cors test.js"
+    "test": "npm run build:umd-polyfill && headless-test --cors test.js",
+    "closure": "google-closure-compiler --js=src/*.js --checks_only --module_resolution=node --compilation_level=ADVANCED",
+    "build": "run-p build:**",
+    "build:cjs": "rollup -i src/index.js -o lib/index.js -f cjs",
```
[the remark this answers is not in the document]
Comment (kuitos) [reply package.json:19]: so your build script can simplified to one line command `"build": "rollup"`

[the remark this answers is not in the document]
Comment (chessl) [reply package.json]: Cool cool, didn't know that. Just updated the rollup config.

File: rollup.config.js [hunk 5]
```diff
@@ -0,0 +1,48 @@
+import { terser } from "rollup-plugin-terser";
+
+export default [
+  {
+    input: 'src/polyfill.js',
+    output: [
+      {
+        file: 'dist/polyfill.js',
+        format: 'umd'
```
Comment (samthor) [inline rollup.config.js:9]: I'm still not convinced anyone actually wants `umd`, but there's probably no harm in it.

I guess I am a bit confused why `umd` doesn't get the document setup stuff that's in index.

**Review by samthor — APPROVED** [review]

**Pull request merged**